### ISSUE TITLE
Keyboard shortcuts are not properly handled on non-qwerty layouts

### ISSUE THREAD

**Issue by euclidianAce** [SerenityOS/serenity#6748]
So I personally use the [colemak](https://colemak.com/) layout (which I was pleasantly surprised was available), but none of the keybindings shown in menus seemed to work.

The problem seems to be that keymaps only change the code point in keypress events and not the key code, and keyboard shortcuts (among other things) seem to only care about the keycode that was received and not the codepoint. So for me, this makes a lot of the functionality unusable.

A short list of things that are affected by this:
 - I tried to remap caps lock to backspace for colemak, which worked fine in the terminal, but in a TextEditor, it would just insert a literal '\b' character and still toggle caps lock
 - Vim emulation in HackStudio still uses qwerty maps (so as both a colemak and vim user, I find it unusable)
 - basically any keyboard shortcut for anything (I think you get the idea)

I hacked some solutions for the remapping of the caps lock key but they are not portable nor scalable and I think this is a more structural problem of how keymaps are implemented that I am too unfamiliar with Serenity (at the time) to really see a good solution for this.

**Comment by euclidianAce** [comment]
Closed via #11023
